### PR TITLE
En 2888

### PR DESCRIPTION
**Pull request by rjmac**
The soql-postgres-adapter half of this thing.

### CODE REVIEW

File: store-pg/src/main/scala/com/socrata/pg/store/events/RowsChangedPreviewHandler.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      if(!truncated) {
+        val copier = pgu.datasetContentsCopier(logger)
+        val schema = pgu.datasetMapReader.schema(newTruthCopyInfo)
+        val copyCtx = new DatasetCopyContext[SoQLType](newTruthCopyInfo, schema)
+        copier.copy(truthCopyInfo, copyCtx)
+      }
+
+      schema.values.find(_.isSystemPrimaryKey).foreach(schemaLoader.makeSystemPrimaryKey)
+
+      pgu.tableDropper.scheduleForDropping(truthCopyInfo.dataTableName)
```
Comment (chitang) [inline store-pg/src/main/scala/com/socrata/pg/store/events/RowsChangedPreviewHandler.scala]: Do we currently interfere with query while the secondary copy is being updated?  Do we need the ability to delay table drop until the playback is complete?  There can be reindexing and rollup at the end.  They can take a long time to process.

Comment (rjmac) [inline store-pg/src/main/scala/com/socrata/pg/store/events/RowsChangedPreviewHandler.scala]: That's why it just schedules it for dropping; this puts the table name in a list, and an asynchronous process comes by later to actually drop it.

**Comment by chitang** [comment]
lgtm if the interference between update and query does not get worse.

File: store-pg/src/main/scala/com/socrata/pg/store/events/RowsChangedPreviewHandler.scala (hunk 1 above)
Comment (chitang) [inline store-pg/src/main/scala/com/socrata/pg/store/events/RowsChangedPreviewHandler.scala]: OK, I think actual clean up already has a buffer.

**Comment by rjmac** [comment]
Hrm, this change has broken pg-secondary tests.  I think it's related to the switch from ordering by `copy_number` to `data_version` in some queries.

**Pull request merged**